tests/test_amex_browseros_bridge.py

```
@@ -0,0 +1,331 @@
+    @patch.object(bridge.subprocess, "run")
+    def test_does_not_restart_listening_browseros(
+        self, run: MagicMock, _endpoint_listening: MagicMock
+    ) -> None:
+        bridge.ensure_browseros_mcp("http://127.0.0.1:9010/mcp")
+
+        run.assert_not_called()
+
+    def test_rejects_non_amex_login_page(self):
+        snapshot = "[UNTRUSTED_PAGE_CONTENT origin=https://example.com/login/]"
+
+        with self.assertRaisesRegex(RuntimeError, "outside americanexpress.com"):
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)